packages/wonder-blocks-announcer/src/__tests__/util/dom.test.ts

```diff
@@ -0,0 +1,133 @@
+import {
+    createRegionWrapper,
+    createDuplicateRegions,
+    createRegion,
+    removeMessage,
+} from "../../util/dom";
+
+jest.useFakeTimers();
+jest.spyOn(global, "setTimeout");
+
+describe("Announcer utility functions", () => {
+    describe("createRegionWrapper", () => {
+        test("it creates a polite region wrapper element", () => {
+            const element = createRegionWrapper("polite");
+
+            expect(element.tagName).toBe("DIV");
+            expect(element.id).toEqual("wbAWrap-polite");
+        });
+
+        test("it creates an assertive region wrapper element", () => {
+            const element = createRegionWrapper("assertive");
+
+            expect(element.tagName).toBe("DIV");
+            expect(element.id).toEqual("wbAWrap-assertive");
+        });
+    });
+
+    describe("createDuplicateRegions", () => {
+        test("it creates multiple polite region elements", () => {
+            const wrapper = document.createElement("div");
+            const dictionary = new Map();
+
+            const regionList = createDuplicateRegions(
+                wrapper,
+                "polite",
+                2,
+                dictionary,
+            );
+
+            expect(regionList.length).toBe(2);
+            expect(regionList[0].id).toBe("wbARegion-polite0");
+            expect(regionList[1].id).toBe("wbARegion-polite1");
+            expect(dictionary.size).toBe(2);
+        });
+
+        test("it creates multiple assertive region elements", () => {
+            const wrapper = document.createElement("div");
+            const dictionary = new Map();
+
+            const regionList = createDuplicateRegions(
+                wrapper,
+                "assertive",
+                2,
+                dictionary,
+            );
+
+            expect(regionList.length).toBe(2);
+            expect(regionList[0].id).toBe("wbARegion-assertive0");
+            expect(regionList[1].id).toBe("wbARegion-assertive1");
+            expect(dictionary.size).toBe(2);
+        });
+    });
+
+    describe("createRegion", () => {
+        test("it creates a polite Live Region element", () => {
+            const dictionary = new Map();
+            const region = createRegion("polite", 0, dictionary);
+
+            expect(region.id).toBe("wbARegion-polite0");
+            expect(region.getAttribute("aria-live")).toBe("polite");
+            expect(region.getAttribute("role")).toBe("log");
+            expect(dictionary.size).toBe(1);
+        });
+
+        test("it creates an assertive Live Region element", () => {
+            const dictionary = new Map();
+            const region = createRegion("assertive", 0, dictionary);
+
+            expect(region.id).toBe("wbARegion-assertive0");
+            expect(region.getAttribute("aria-live")).toBe("assertive");
+            expect(region.getAttribute("role")).toBe("log");
+            expect(dictionary.size).toBe(1);
+        });
```

Comment:
**suggestion:** Given another look at these tests that only change the politeness level. Maybe you could also add dynamic tests to avoid repeating the whole test structure?

```
test.each(["polite", "assertive"])("it creates an %s Live Region element", (politenessLevel) => {
            // Arrange
            const dictionary = new Map();

            // Act
            const region = createRegion(politenessLevel, 0, dictionary);

            // Assert
            expect(region.id).toBe(`wbARegion-${politenessLevel}0`);
            expect(region.getAttribute("aria-live")).toBe("politenessLevel");
            expect(region.getAttribute("role")).toBe("log");
            expect(dictionary.size).toBe(1);
        });
```

Reply:
Another suggestion would be to split the assertions into multiple tests, so it would help understand better which assertion/test is exactly failing (if that's ever the case), which follows the AAA pattern.